app/assets/javascripts/Components/dashboard.jsx

```diff
@@ -4,6 +4,85 @@ import { render } from 'react-dom';
 import { Bar } from 'react-chartjs-2';
 
 
+class Assignment extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      summary: {
+        average: null,
+        median: null,
+        num_submissions_collected: null,
+        num_submissions_graded: null,
+        num_fails: null,
+        num_zeros: null,
+        groupings_size: null
+      },
+      data: { // Fake data, from react-chartjs-2/example/src/charts/GroupedBar.js
+        labels: ['1', '2', '3', '4', '5', '6'],
+        datasets: [
+          {
+            label: '# of Red Votes',
+            data: [12, 19, 3, 5, 2, 3],
+            backgroundColor: 'rgb(255, 99, 132)',
+          },
+          {
+            label: '# of Blue Votes',
+            data: [2, 3, 20, 5, 1, 4],
+            backgroundColor: 'rgb(54, 162, 235)',
+          },
+          {
+            label: '# of Green Votes',
+            data: [3, 10, 13, 15, 22, 30],
+            backgroundColor: 'rgb(75, 192, 192)',
+          },
+        ],
+      }
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevProps.assessment_id !== this.props.assessment_id) {
```

Comment:
I think you'll want to pull this out into a separate `fetchData` method that can be called from both `componentDidUpdate` and `componentDidMount` (so that it's triggered on an initial mounting of the component).